From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Praschl <praschlpaul@g-p.at>
Date: Sun, 2 Aug 2026 12:14:44 +0200
Subject: [PATCH 069/117] backlight: apple_pmic_bl: return error for unknown
 PMIC types

Return -EINVAL when brightness is requested for an unsupported PMIC type
instead of falling through without a defined result.

Signed-off-by: Paul Praschl <praschlpaul@g-p.at>

Orchard-replay-of: 277dfe010ec0adfd0c9780601d62ae1b54c2ada9
---
 drivers/video/backlight/apple_pmic_bl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/backlight/apple_pmic_bl.c b/drivers/video/backlight/apple_pmic_bl.c
index 64d472512..ae891a6e5 100644
--- a/drivers/video/backlight/apple_pmic_bl.c
+++ b/drivers/video/backlight/apple_pmic_bl.c
@@ -65,6 +65,8 @@ static int apple_pmic_bl_get_brightness(struct backlight_device *bl)
 		case PMIC_TYPE_ARIA:
 			return ((cmd[1] & 7) << 8) | (cmd[0] & 0xff);
 	}
+
+	return -EINVAL;
 }
 
 static const struct backlight_ops apple_pmic_bl_ops = {
